### ISSUE TITLE
How do you run Talos on train on batch data eg. datagenerator

### ISSUE THREAD

**Issue by ruojol** [autonomio/talos#92]
Hello, how do you run Talos with a datagenerator?
I train my model via this loop: 
`or epoch in range(n_epochs):
        user_input, item_input, labels = data_generator(train, p['n_negative'])
        X = [np.array(user_input), np.array(item_input)]
        y = np.array(labels)
        history = model.fit(X,
                            y,
                            epochs=n_epochs,
                            shuffle=True,
                            validation_split=0.2)`
So do I just run talos in a loop like that or how is it done?
eg.
`for epoch in range(n_epochs):
    user_input, item_input, labels = data_generator(train, p['n_negative'])
    X = [np.array(user_input), np.array(item_input)]
    y = np.array(labels)
    t = ta.Scan(x=X,
                y=y,
                val_split=0.3,
                model=DAE_CF,
                params=p,
                dataset_name='movie_lens_test',
                experiment_no='1',
                experimental_functional_support=True)`

**Comment by mikkokotila** [comment]
Is it related with #11 and/or #67? 

If you run Talos inside a loop, it basically becomes a loop inside a loop i.e. you are going to have run n experiments for every step in the outer loop you are creating.

**Comment by ruojol** [comment]
Yeah it's related so I'll just wait for the update or is it available on daily-dev?

**Comment by mikkokotila** [comment]
Thanks. Keep an eye (or subscribe to) #11. Closing here.

**Comment by ruojol** [comment]
Actually now that I've given it some study it isn't the same exact issue. I'm not doing training on a fit_generator function but rather a train on batch loop, but I can propably reformat the code to use a fit_generator since it's in the roadmap.
ps. tuning hyperparameters for a multi-input model turns out to be a real pain in the ass since Talos is the only package I've found thas supports the keras-functional API....
